README.md

```diff
@@ -1,488 +1,57 @@
-    * `IGN_TEST_JWT`
-
-
-# Run the backend server
-
-First, make sure to set the `AUTH0_RSA256_PUBLIC_KEY` environment variable with the Auth0 RSA256 public key. This env var will be used by the backend to decode and validate any received Auth0 JWT tokens.
-Note: You can get this key from: <https://osrf.auth0.com/.well-known/jwks.json> (or from your own auth0 user). Open that url in the browser and copy the value of the `x5c` field.
-
-```
-$GOPATH/bin/web-cloudsim
-```
-
-
-## Cloudsim optional Env vars (in the .env file)
-
-- export IGN_CLOUDSIM_SSL_PORT=(default 4431)
-- export IGN_CLOUDSIM_HTTP_PORT=(default 8001)
-
-Also, if using Kubernetes with EC2:
-
-- export KUBEADM_JOIN=kubeadm join IP:PORT --token ...
-- export KUBE_MASTER_IP=the ip of the master node.
-- export IGN_TRANSPORT_IP_INTERFACE=`ethwe` (if using Weave Network in the kubernetes cluster)
-- export AWS_GZ_LOGS_BUCKET=a bucket name to upload gazebo logfiles.
-
-Note: To backup GZ logs into S3 it will also require a `kubernetes secret` called `aws-secrets` in the cluster.
-That secret should at least contain the following `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` entries.
-You can create the secret by running the script `create-secrets.sh` included with the web-cloudsim code.
-These are the contents of that script:
-
-```
-#/bin/bash
-
-# This script is used to create a Kubernetes secret with name 'aws-secrets' based on existing
-# environment variables: AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY.
-# This Secret will be used by the cloudsim server.
-
-kubectl create secret generic aws-secrets --from-literal=aws-access-key-id=${AWS_ACCESS_KEY_ID} --from-literal=aws-secret-access-key=${AWS_SECRET_ACCESS_KEY}
-```
-
-
-Note2: to find the value for the KUBEADM_JOIN env variable, from the kubernetes master, run:
-`sudo kubeadm token create --ttl 0 --print-join-command` and copy the results.
-Read the k8-readme.md for more info.
-
-
-
-# Kubernates
-The cloudsim server will try to connect to a Kubernates master. To do this, it will look
-for the following config file: `<home>/.kube/config`.
-
-
-For more details and tips about kubernetes, see the `k8-readme.md` file.
-
-
-
-# Linter
-
-1. Get the linter (gometalinter)
-
-    ```
-    cd ~/go_ws
-    ```
+[![Format](https://img.shields.io/github/actions/workflow/status/gazebo-web/cloudsim/format.yaml?branch=main&label=Format)](https://github.com/gazebo-web/cloudsim/actions?query=workflow:"Format")
+[![Test](https://img.shields.io/github/actions/workflow/status/gazebo-web/cloudsim/test.yaml?branch=main&label=Test)](https://github.com/gazebo-web/cloudsim/actions?query=workflow:"Test")
+</div>
 
-    ```
-    curl -L https://git.io/vp6lP | sh
-    ```
+## Getting started
 
-1. Run the linter
+The cloudsim library provides multiple components that are used by applications to run simulations in the cloud. These components usually are an abstraction of a third-party service that will be consumed by cloudsim, and they were created in
```

Comment:
I think it's important to define a vocabulary before jumping into things. We are talking about applications and components without having introduced those concepts beforehand. 

We should have an intro that explains that this repo contains Cloudsim, which is a library that provides a set of components that applications can use to run robotic applications in the cloud. 

We should also move the **Features** section before this section to explain that there are different component types, and that each component type typically has a 1:1 relationship with some established technology (e.g. Kubernetes) or provider (e.g. AWS, GCP, Azure, etc.). Listing current implementations of components might be helpful for people to understand both the component's role and Cloudsim's capabilities.

Reply:
> We should have an intro that explains that this repo contains Cloudsim, which is a library that provides a set of components that applications can use to run robotic applications in the cloud.

That's already added a as description to the logo, I'm trying to write something more meaningful but I can't find anything to add other than a brief definition of what Applications and Components are, which was already kind of explained in the text highlighted by your comment.

> We should also move the Features section before this section to explain that there are different component types, and that each component type implementation typically has a 1:1 relationship with some established technology (e.g. Kubernetes) or provider (e.g. AWS, GCP, Azure, etc.). Listing current implementations of components might be helpful for people to understand both the component's role and Cloudsim's capabilities.

This could be included as a definition of a component, but we should probably add a table to show what are the current implementations for the different components, although this was made pretty obvious now that we follow the `pkg/<component>/implementation/<technology>` folder structure.

Reply:
This tells me we should invest a bit more of time into our source code documentation to implement the latest changes from `go doc`: https://go.dev/doc/comment

Reply:
We can definitely do that in another PR. Feel free to open an issue for that.

Reply:
https://github.com/gazebo-web/cloudsim/issues/9